mainloop.cpp

```diff
@@ -0,0 +1,433 @@
+/*
+ * This file is part of the MAVLink Router project
+ *
+ * Copyright (C) 2016  Intel Corporation. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "mainloop.h"
+
+#include <assert.h>
+#include <memory>
+#include <signal.h>
+#include <sys/epoll.h>
+#include <unistd.h>
+
+#include "log.h"
+
+static bool should_exit = false;
```

Comment:
I thought it was ok since I was reviewing the idea, not the changes... it would only move code around, but then I went to check this one thing and it looked wrong.

It was `static volatile bool g_should_exit;` before and you removed the volatile.